OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/Optimove.java

```diff
@@ -144,25 +130,64 @@ public static void initialize(@NonNull Application application, @NonNull Optimov
         }
 
         if (config.isOptimoveConfigured()) {
-            if (config.getCustomMinLogLevel() != null) {
-                OptiLoggerStreamsContainer.setMinLogLevelToShow(config.getCustomMinLogLevel());
-            }
+            Optimove.finishInitialization(application, config);
+        }
+    }
 
-            OptiLoggerStreamsContainer.initializeLogger(application.getApplicationContext());
-
-            Runnable initCommand = () -> {
-                OptiLoggerStreamsContainer.debug("Optimove.initialize() is starting");
-                shared.lifecycleObserver.addActivityStoppedListener(shared.optimoveLifecycleEventGenerator);
-                shared.lifecycleObserver.addActivityStartedListener(shared.optimoveLifecycleEventGenerator);
-                application.registerActivityLifecycleCallbacks(shared.lifecycleObserver);
-                shared.fetchConfigs();
-            };
-            if (!OptiUtils.isRunningOnMainThread()) {
-                OptiLoggerStreamsContainer.debug("Optimove.initialize() was called from a worker thread, moving call to main thread");
-                OptiUtils.runOnMainThread(initCommand);
-            } else {
-                initCommand.run();
-            }
+    private static void finishInitialization(@NonNull Application application, @NonNull OptimoveConfig config){
```

Comment:
Is this specific for the T&T/Optimove side of the SDK? If so, perhaps renaming finishOptimoveInit or similar?